### ISSUE TITLE
random_choice missing from fable library

### ISSUE THREAD

**Issue by joprice** [fable-compiler/Fable#3966]
### Description

A runtime error is hit when trying to use `Seq.randomChoice`: 

```
ImportError: cannot import name 'random_choice' from 'fable_modules.fable_library.seq'
```

It looks like this is a group of relatively recent helpers from this rfc (https://github.com/fsharp/fslang-design/blob/main/RFCs/FS-1135-random-functions-for-collections.md), and need to be added to the replacements:
https://github.com/dotnet/fsharp/commit/54a19976c4529286b360dace622060b0ce5317aa#diff-e145f82a233516da994063cddbbe920e616e734ee6ab652eb4c8a21f76382ef9

### Related information

* Fable version: `4.23.0`
* Operating system: OSX

**Comment by joprice** [comment]
Yes I expected the typical unsupported error as well. I guess to handle that every supported method needs to be listed out in the replacements to avoid new passing through.

**Comment by MangelMaxime** [comment]
>  I guess to handle that every supported method needs to be listed out in the replacements to avoid new passing through.

Indeed, I often think about if we should or should not do it.

On one side, have an exhaustive list of the support API make the DX better because Fable can capture the error.

On the other side, it makes adding a new API a bit more verbose and also means that for every API call Fable will need to check the list of supporting API. We would probably not want to use a list for that but a `Set` or `Hashmap` if I remember correctly to get a `O(1)` search.

IHMO I think the pros are worth considering.

**Comment by joprice** [comment]
Yes, luckily it results in a hard error that prevents the script from executing, but having a compile time error benefits library development as well to find unsupported usages without requiring full coverage.

It could end up being a lot of string comparisons in a large codebase. A hash or treemap/set lookup or even a trie could be used to optimize it.
